cpp/src/arrow/stl_test.cc

```diff
@@ -500,5 +503,60 @@ TEST(TestTupleVectorFromTable, CastingNeeded) {
   ASSERT_EQ(rows, expected_rows);
 }
 
+TEST(STLMemoryPool, Base) {
+  std::allocator<uint8_t> allocator;
+  STLMemoryPool<std::allocator<uint8_t>> pool(allocator);
+
+  uint8_t* data = nullptr;
+  ASSERT_OK(pool.Allocate(100, &data));
+  ASSERT_EQ(pool.max_memory(), 100);
+  ASSERT_EQ(pool.bytes_allocated(), 100);
+  ASSERT_NE(data, nullptr);
+
+  ASSERT_OK(pool.Reallocate(100, 150, &data));
+  ASSERT_EQ(pool.max_memory(), 150);
+  ASSERT_EQ(pool.bytes_allocated(), 150);
+
+  pool.Free(data, 150);
+
+  ASSERT_EQ(pool.max_memory(), 150);
+  ASSERT_EQ(pool.bytes_allocated(), 0);
+}
+
+TEST(stl_allocator, MemoryTracking) {
```

Comment:
using `_` in GTest names can lead to subtle collisions [and is specifically not recommended.](https://github.com/google/googletest/blob/master/googletest/docs/faq.md#why-should-test-suite-names-and-test-names-not-contain-underscore) You didn't introduce this but could you please fix this by removing that `_`? (If you rename `stl_allocator` to `allocator` then we won't have the problem)